src/MultiSearchPrompt.php

```diff
@@ -0,0 +1,233 @@
+<?php
+
+namespace Laravel\Prompts;
+
+use Closure;
+use Illuminate\Support\Collection;
+
+class MultiSearchPrompt extends Prompt
+{
+    use Concerns\ReducesScrollingToFitTerminal;
+    use Concerns\Truncation;
+    use Concerns\TypedValue;
+
+    /**
+     * The index of the highlighted option.
+     */
+    public ?int $highlighted = null;
+
+    /**
+     * The index of the first visible option.
+     */
+    public int $firstVisible = 0;
+
+    /**
+     * The cached matches.
+     *
+     * @var array<int|string, string>|null
+     */
+    protected ?array $matches = null;
+
+    /**
+     * The default values the multi-search prompt.
+     *
+     * @var array<int|string, string>
+     */
+    public array $default;
+
+    /**
+     * The selected values.
+     *
+     * @var array<int|string, string>
+     */
+    public array $values = [];
+
+    /**
+     * Create a new MultiSearchPrompt instance.
+     *
+     * @param  array<int|string, string>|Collection<int|string, string>  $default
+     * @param  Closure(string): array<int|string, string>  $options
+     */
+    public function __construct(
+        public string $label,
+        public Closure $options,
+        public bool $returnKeys = true,
+        array|Collection $default = [],
```

Comment:
The `default` behaviour is a little confusing and perhaps unexpected. At first, I didn't realise I needed to pass an associative array, but it makes sense because we don't have the resolved options yet to retrieve a label as we do for the `multiselect` prompt. However, everywhere else, the type passed to `default` matches the return type of the prompt.

Additionally, if you uncheck the default value and then perform a search, it's no longer possible to find and re-select the default value unless you ensure it appears in your search results.

I'm not opposed to including a `default`, but I think we should leave it out for now and revisit it in a separate PR, potentially adding the same functionality to the `search` prompt at the same time.

Reply:
Perhaps you could pass `default` as just an array of keys and then ensure that your `options` callback includes those items on the first render (i.e. when the search value is empty). The same could be done for the `search` prompt.